src/compiler/parsing/watf/tokenizer.js

```diff
@@ -133,7 +133,7 @@ function tokenize(input: string) {
     }
 
     const NUMBERS = /[0-9|.|_]/;
-    const HEX_NUMBERS = /[0-9|A-F|a-f|_|.]/;
+    const HEX_NUMBERS = /[0-9|A-F|a-f|_|.|p|P|-]/;
```

Comment:
How do you parse `nan:0x40000`? Could we add a test for this case?

Reply:
These are not included in here, I am working on it in a separate branch:

https://github.com/xtuc/js-webassembly-interpreter/tree/nan-payloads

I planned to fix all merge conflicts before PRing from the nan-payloads branch and do this separately.

Reply:
ok